Thanks for figuring this out. Offering some quick confirmation of the basic idea here:

```
$ mkdir /tmp/awx_1234; mkdir /tmp/awx_4321; mkdir /tmp/awx_2345
$ ansible-runner worker cleanup --file-pattern=/tmp/awx_* --exclude-strings="1234, 2345" --grace-period=0
Removed 3 private data dir(s) in pattern /tmp/awx_*
(changed: True)
```

Removing all 3 directories is unintended here, because they were passed in the `--exclude`. Your fix:

```
$ mkdir /tmp/awx_1234; mkdir /tmp/awx_4321; mkdir /tmp/awx_2345
$ ansible-runner worker cleanup --file-pattern=/tmp/awx_* --exclude-strings 1234 2345 --grace-period=0
Removed 1 private data dir(s) in pattern /tmp/awx_*
(changed: True)
```

Out of the 3 folders, this commands excludes 2 of them, which is intended. It should still delete the folder not in the `--exclude` which is the 4321 folder. That works as intended.

My only very minor comment / request-for-change would be from looking at your test output:

```
AssertionError: assert 'cleanup --re...utable=podman' == 'cleanup --re...utable=podman'
  
  - cleanup --remove-images="quay.invalid/foo/bar:latest quay.invalid/foo/bar:devel" --image-prune --process-isolation-executable=podman
  ?                        ^^                                                      -
  + cleanup --remove-images quay.invalid/foo/bar:latest quay.invalid/foo/bar:devel --image-prune --process-isolation-executable=podman
  ?    
```

IMO, a better standard pattern would be to retain the quotes, but for each arg. Getting rid of the `=` also seems necessarily. This won't make any sense until I form the command here:

```
$ mkdir /tmp/awx_1234; mkdir /tmp/awx_4321; mkdir /tmp/awx_2345
$ ansible-runner worker cleanup --file-pattern "/tmp/awx_*" --exclude-strings "1234" "2345" --grace-period=0
Removed 1 private data dir(s) in pattern /tmp/awx_*
(changed: True)
```

This still gives us the desired behavior of deleting the 4321 directory. But for _all_ arguments, we ditch the `=` and add `"`, the important nuance is that for multil-args, we put `"` around each positional entry. And actually, the `*` won't parse correctly if you don't have quotes, so I think keeping quotes is still best as a universal policy.

### Consequence of bug

The cleanup job needs to use the `--exclude` to prevent the private_data_dir of actively _running_ jobs from being deleted while that job is running. With >1 job running on a given node, this would seem to break the exclusion, while still allowing the cleanup to run, meaning a private_data_dir for a running job may (incorrectly) get deleted, obviously disrupting the job.

As you mention "JSON error", I want to talk this out that we're talking about errors in jobs that just happen to be running at the same time the cleanup command runs, because of disruptions from their private_data_dir being deleted. That is generally what I would expect from this.

--- a/awx/main/tasks/receptor.py
+++ b/awx/main/tasks/receptor.py
@@ -360,7 +360,7 @@ def _convert_args_to_cli(vargs):
     args = ['cleanup']
     for option in ('exclude_strings', 'remove_images'):
         if vargs.get(option):
-            args.append('--{} {}'.format(option.replace('_', '-'), ' '.join(vargs.get(option))))
+            args.append('--{} {}'.format(option.replace('_', '-'), ' '.join(f'"{item}"' for item in vargs.get(option))))
     for option in ('file_pattern', 'image_prune', 'process_isolation_executable', 'grace_period'):
         if vargs.get(option) is True:
             args.append('--{}'.format(option.replace('_', '-')))

--- a/awx/main/tasks/receptor.py
+++ b/awx/main/tasks/receptor.py
@@ -360,7 +360,7 @@ def _convert_args_to_cli(vargs):
     args = ['cleanup']
     for option in ('exclude_strings', 'remove_images'):
         if vargs.get(option):
-            args.append('--{}="{}"'.format(option.replace('_', '-'), ' '.join(vargs.get(option))))
+            args.append('--{} {}'.format(option.replace('_', '-'), ' '.join(vargs.get(option))))
     for option in ('file_pattern', 'image_prune', 'process_isolation_executable', 'grace_period'):
         if vargs.get(option) is True:
             args.append('--{}'.format(option.replace('_', '-')))

--- a/awx/main/tests/unit/utils/test_receptor.py
+++ b/awx/main/tests/unit/utils/test_receptor.py
@@ -3,7 +3,7 @@ from awx.main.tasks.receptor import _convert_args_to_cli
 
 def test_file_cleanup_scenario():
     args = _convert_args_to_cli({'exclude_strings': ['awx_423_', 'awx_582_'], 'file_pattern': '/tmp/awx_*_*'})
-    assert ' '.join(args) == 'cleanup --exclude-strings="awx_423_ awx_582_" --file-pattern=/tmp/awx_*_*'
+    assert ' '.join(args) == 'cleanup --exclude-strings awx_423_ awx_582_ --file-pattern=/tmp/awx_*_*'
 
 
 def test_image_cleanup_scenario():

--- a/awx/main/tests/unit/utils/test_receptor.py
+++ b/awx/main/tests/unit/utils/test_receptor.py
@@ -3,7 +3,7 @@ from awx.main.tasks.receptor import _convert_args_to_cli
 
 def test_file_cleanup_scenario():
     args = _convert_args_to_cli({'exclude_strings': ['awx_423_', 'awx_582_'], 'file_pattern': '/tmp/awx_*_*'})
-    assert ' '.join(args) == 'cleanup --exclude-strings awx_423_ awx_582_ --file-pattern=/tmp/awx_*_*'
+    assert ' '.join(args) == 'cleanup --exclude-strings "awx_423_" "awx_582_" --file-pattern=/tmp/awx_*_*'
 
 
 def test_image_cleanup_scenario():
@@ -17,5 +17,6 @@ def test_image_cleanup_scenario():
         }
     )
     assert (
-        ' '.join(args) == 'cleanup --remove-images="quay.invalid/foo/bar:latest quay.invalid/foo/bar:devel" --image-prune --process-isolation-executable=podman'
+        ' '.join(args)
+        == 'cleanup --remove-images "quay.invalid/foo/bar:latest" "quay.invalid/foo/bar:devel" --image-prune --process-isolation-executable=podman'
     )

